src/install-pnpm/run.ts

```diff
@@ -21,7 +23,18 @@ export async function runSelfInstaller(inputs: Inputs): Promise<number> {
 
   // prepare target pnpm
   const target = await readTarget({ version, packageJsonFile, standalone })
-  const cp = spawn(execPath, [path.join(__dirname, 'pnpm.cjs'), 'install', target, '--no-lockfile'], {
+  const installArgs = [path.join(__dirname, 'pnpm.cjs'), 'install', target, '--no-lockfile']
+
+  if (GITHUB_WORKSPACE) {
+    const customRegistry = readNpmrcRegistry(GITHUB_WORKSPACE)
```

Comment:
Looks like the consuming function is already async, so awaiting here _should_ be OK:
```suggestion
    const customRegistry = await readNpmrcRegistry(GITHUB_WORKSPACE)
```